### ISSUE TITLE
amp-instagram re-render when above the fold

### ISSUE THREAD

**Issue by TWsky** [ampproject/amphtml#17122]
## What's the issue?

My partner has some UX issue while they're using < amp-instagram >.

As video [here](https://drive.google.com/open?id=1wbLpnyG77fiuFNfMvJJmHkb4V-ya1yNf) can see that amp-instagram draws the photo frame after it is above the fold, and it would make the UI re-flow + content position changed.

It's not because of giving the wrong width & height, since I've tried it and it still happens the same.

It would more like to be a bug since AMP shouldn't let layout jump above the fold.

## How do we reproduce the issue?

1. go to [here](http://playing.ltn.com.tw/amp/article/10117)
2. scroll down the page and can see the instagram section re-flow

## What browsers are affected?

Tested on Chrome 67.0.3396.99

## Which AMP version is affected?

Version 153180087910

**Comment by choumx** [comment]
I think this is working as intended. amp-instagram, along with a few other components like amp-twitter, do allow changing height in ways that can cause content jumping. This is because there's no good way to determine the height of the caption a priori.

The jumping may be reduced if `data-captioned` is removed, but not sure if that's acceptable to the partner.

**Comment by choumx** [comment]
Per the comments on amp-instagram.js, try setting `width="320" height="392" layout="responsive"`.

https://github.com/ampproject/amphtml/blob/d01e111c946c1f61bb2896df0b8351f0cb9f5117/extensions/amp-instagram/0.1/amp-instagram.js#L20-L26

**Comment by TWsky** [comment]
Tried [here](https://amp-spa.firebaseapp.com/ampInstagram/), still re-flow during above the fold.

Seem to be that layout="responsive" works great which helps the amp-instagram fits the layout, but the problem would still be the height recalculating before the instagram portion has been settle down.

**Comment by choumx** [comment]
Looking through the code history, it looks like that example may be out of date (e.g. Instagram's embed UI may have changed since). We may be able to implement a better placeholder for amp-instagram or at least figure out how to avoid reflow, since it the frame modulo caption appears to be statically sized.

/to @aghassemi

**Comment by cramforce** [comment]
yes, this is a polish bug on the `amp-instagram` plugin. Basically the outside layer hardcodes certain offsets to position the placeholder like they would be inside the plugin, and that breaks when instagram changes only the plugin. Should be easy to fix.

Ideally, insta would do this themselves when they change iframe layout. CC @adewale

**Comment by aghassemi** [comment]
related: https://github.com/ampproject/amphtml/issues/17570
